### PR TITLE
add regex for space+ignoreList for valid ws names

### PR DESCRIPTION
**Pull request by sourcecodedeveloper**
### Fixed Issues

$ https://github.com/Expensify/App/issues/23376
PROPOSAL: https://github.com/Expensify/App/issues/23376#issuecomment-1676451184

### Tests

1. Go to Settings -> Workspaces -> Click on existing workspace or create new
2. From workspace click settings and then edit name
3. For test cases, following workspace names have been added to ignorelist and test valid.
    a. `<>`
    b. `< >`
    c. `<->`
    d. `<-->`
    e. `<br>`
    f. `<br/>`
4. Testing each of the valid workspace names should pass validation on front end.
5. Invalid workspace names include html tags with trailing spaces and any other html tag not included in the ignore list. Entering the following workspace names should fail validation.
    a. `< lol>`
    b. `<>< ><--><-><br><br/><lol>< lol>`
    c. `<lol>< lol><>< ><--><-><br><br/>`
    d. `<--><br><>< ><br><lol>`
    e. `<>< lol>`
    f. `<>< ><br >`
6. Valid workspace names. Entering the following workspace names should pass validation.
    a. `<>< ><--><-><br><br/>`
    b. `<->workspace name<->`
    c. `<br>workspace name<br/>`
    d. `<--><br><>< ><br>`
- [x] Verify that no errors appear in the JS console

### Offline tests

PR changes not affected by network

### QA Steps

QA steps listed in Tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

[workspacenames-web.webm](https://github.com/Expensify/App/assets/141340971/191d5a5e-23a9-44fe-a9f3-d02fdaf53742)
</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/141340971/25168029-8090-4984-aec9-c595b9ed7c9b

</details>

<details>
<summary>Mobile Web - Safari</summary>

[workspacenames-ios-web.webm](https://github.com/Expensify/App/assets/141340971/66fb49b1-c175-4109-b25d-f93b6b31d0aa)

</details>

<details>
<summary>Desktop</summary>

[workspacenames-desktop-app.webm](https://github.com/Expensify/App/assets/141340971/a3db5ef0-7dd9-43bf-be95-6f673e9d42d4)

</details>

<details>
<summary>iOS</summary>

[workspacenames-ios-app.webm](https://github.com/Expensify/App/assets/141340971/4a51a568-f17d-4e6c-9057-46b621191b23)

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/141340971/0eb510ff-27dd-489e-ac45-663a25af71f8
</details>

### CODE REVIEW

File: src/components/Form.js
Comment (robertKozik) [inline src/components/Form.js:154]:
```suggestion
                        if (!isMatch) {
```

File: src/components/Form.js
Comment (robertKozik) [inline src/components/Form.js:152]: Can we get more meaningful name for that? Like `tag` or `htmlTag` 🤔

File: src/components/Form.js
Comment (robertKozik) [inline src/components/Form.js:160]:
```suggestion
                if (isMatch && leadingSpaceIndex === -1) {
```

File: src/components/Form.js
Comment (robertKozik) [inline src/components/Form.js:140]: Coming from [this discussion](https://github.com/Expensify/App/pull/25530#discussion_r1308736503). I think we can easily move these two lines right before the 149 line 
```js
if (matchedHtmlTags) {
```

**Review by robertKozik — CHANGES_REQUESTED** [review]

File: src/components/Form.js
Comment (sourcecodedeveloper) [reply src/components/Form.js:152]: better to just drop the const and use matchedHtmlTags[i] that way don't have to worry about coming up with more meaningful name.

Comment (sourcecodedeveloper) [reply src/components/Form.js]: I went ahead and changed the name to htmlTag! changes pushed.

**Comment by sourcecodedeveloper** [comment]
> CC. @techievivek as you are not assigned to this PR yet (It's not approved by me - but the former one was)

@robertKozik @techievivek PR changes resolved. some workflows are awaiting approval.

**Comment by sourcecodedeveloper** [comment]
@robertKozik some checks are waiting on PR Reviewer checklist please add it. Hoping that the PR can be merged soon if the checks pass.

**Comment by sourcecodedeveloper** [comment]
@robertKozik @techievivek please kindly expedite the reviewing, worried about the 50% **penalty**

**Review by techievivek — APPROVED** [review]
Looks good on my end. I will wait for Robert to finish up the checklist.

**Comment by robertKozik** [comment]
On it 👀

**Comment by robertKozik** [comment]
@sourcecodedeveloper whitelisted tags are ignored, except `<>` `< >`. The validation should pass for whitelisted tags. Video attached

https://github.com/Expensify/App/assets/61146594/025a227b-a460-4706-bea9-a709b0e46a00

**Comment by sourcecodedeveloper** [comment]
> 



> @sourcecodedeveloper whitelisted tags are ignored, except `<>` `< >`. The validation should pass for whitelisted tags. Video attached
> Screen.Recording.2023-08-30.at.12.03.47.mov

@robertKozik the WHITE got missed from const.js lol, re-added WHITELISTED_TAGS to const.js and the tests are passing.

**Comment by sourcecodedeveloper** [comment]
> white



> @sourcecodedeveloper whitelisted tags are ignored, except `<>` `< >`. The validation should pass for whitelisted tags. Video attached
> Screen.Recording.2023-08-30.at.12.03.47.mov



> @sourcecodedeveloper whitelisted tags are ignored, except `<>` `< >`. The validation should pass for whitelisted tags. Video attached
> Screen.Recording.2023-08-30.at.12.03.47.mov

@techievivek @robertKozik Please expedite reviewing.

**Comment by sourcecodedeveloper** [comment]
> Looks good on my end. I will wait for Robert to finish up the checklist.

Thanks but the review got stale.

**Comment by robertKozik** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by techievivek — APPROVED** [review]
Looks good. Thanks

**Comment by robertKozik** [comment]
@techievivek as we have now **merge freeze :cold_face: for PRs** ([info](https://expensify.slack.com/archives/C01GTK53T8Q/p1693347701566439)), I think we should wait with the merge here. I think is not that urgent fix

**Comment by sourcecodedeveloper** [comment]
@robertKozik Workflow checks have passed, just waiting for your approval.

**Comment by techievivek** [comment]
Ah, great point @robertKozik, I think this can wait and @sourcecodedeveloper this wouldn't result in penalty as we have officially approved the PR but holding it since there is a merge freeze going on.

**Comment by techievivek** [comment]
Added the HOLD so we don't merge this by mistake.

**Comment by techievivek** [comment]
@sourcecodedeveloper Can you please follow the instructions from here https://expensify.slack.com/archives/C01GTK53T8Q/p1694405959277169, i.e. update the PR with the latest main and re-test it? Thanks

**Comment by sourcecodedeveloper** [comment]
> @sourcecodedeveloper Can you please follow the instructions from here https://expensify.slack.com/archives/C01GTK53T8Q/p1694405959277169, i.e. update the PR with the latest main and re-test it? Thanks
@techievivek PR updated with latest main and tested as well.

File: src/components/Form.js
Comment (techievivek) [reply src/components/Form.js:143]: I see. I think the PR was not tested for Bank account flow. Thanks for reporting it.

Comment (techievivek) [reply src/components/Form.js:143]: @sourcecodedeveloper Could you figure out a fix for this so we can CP the changes otherwise we will need to revert the PR.

Comment (techievivek) [reply src/components/Form.js:143]: We can probably use the check `isString` that we use at line 126.

```
const trimmedStringValues = {};
_.each(values, (inputValue, inputID) => {
    if (_.isString(inputValue)) {
        trimmedStringValues[inputID] = inputValue.trim();
    } else {
        trimmedStringValues[inputID] = inputValue;
    }
});
```

Comment (techievivek) [reply src/components/Form.js:143]: Let me push a PR.

Comment (sourcecodedeveloper) [reply src/components/Form.js:143]: @robertKozik @techievivek I retested the test cases along with the bank account flow and the [PR](https://github.com/Expensify/App/pull/27217) submitted is working as expected, i.e no more crash when clicking on checkbox.

**Pull request merged**
